src/python/WMCore/WMSpec/WMWorkloadTools.py

```diff
@@ -153,13 +127,18 @@ def _validateArgumentOptions(arguments, argumentDefinition, optionKey):
 
 
 def _validateInputDataset(arguments):
+    
     inputdataset = arguments.get("InputDataset", None)
     dbsURL = arguments.get("DbsUrl", None)
     if inputdataset != None and dbsURL != None:
-        result = _verifyDBSCall(dbsURL, "datasets?&dataset_access_type=*&dataset=%s" % inputdataset)
-        if len(result) == 0:
-            msg = "Inputdataset %s doesn't exist on %s" % (inputdataset, dbsURL)
-            raise WMSpecFactoryException(msg)
+        #import DBS3Reader here, since Runtime code import this module and worker node doesn't have dbs3 client 
+        from WMCore.Services.DBS.DBS3Reader import DBS3Reader
+        from WMCore.Services.DBS.DBSErrors import DBSReaderError
+        try:
```

Comment:
I don't follow it. Why we want to contact DBS from the worker node? Or are you referring to the jenkins node that does not have dbs client?
Sorry if I missed something, but can't we mock these checks instead of adding these imports in the middle of the code?